### ISSUE TITLE
'Invoke-WebRequest -SkipCertificateCheck' doesn't work with netcoreapp2.0 on Mac

### ISSUE THREAD

**Issue by daxian-dbw** [PowerShell/PowerShell#3648]
This is because `HttpClientHandler.ServerCertificateCustomValidationCallback` currently doesn't work in `netcoreapp2.0` on Mac at all. See the [comment](https://github.com/dotnet/corefx/issues/9728#issuecomment-297186983) in https://github.com/dotnet/corefx/issues/9728 for more information.

Note that we currently have 2 tests failing in nightly run on OSX CI:
```
Validate Invoke-WebRequest -SkipCertificateCheck
Validate Invoke-RestMethod -SkipCertificateCheck
```
Error message is:
```
The handler does not support custom handling of certificates with this combination of libcurl (7.49.1) and its SSL backend ("SecureTransport").
```

**Comment by SteveL-MSFT** [comment]
@joeyaiello we should add this to Known Issues

**Comment by joeyaiello** [comment]
@daxian-dbw it looks like dotnet/corefx#19709 would solve our problem as well. Could you validate that?

**Comment by stephentoub** [comment]
Please let us know if https://github.com/dotnet/corefx/pull/19941 doesn't address this need.  I believe it should.

**Comment by iSazonov** [comment]
@daxian-dbw  Is the Issue fixed?

**Comment by daxian-dbw** [comment]
Yes, it's fixed by #3887

**Comment by daxian-dbw** [comment]
We are moving back to .NET Core 2.0.0-preview1 via PR #4026 due to regressions in the latest .NET Core 2.0.0-preview2 (dotnet/corefx#21095). So re-open this bug.

**Comment by iSazonov** [comment]
@daxian-dbw It seems CoreFX was fixed and we can move to new version.

**Comment by daxian-dbw** [comment]
@iSazonov I would love to, but the `2.0.0-preview3` .NET CLI is broken: https://github.com/dotnet/cli/issues/7013

**Comment by iSazonov** [comment]
😄 Again?! It looks we near release candidate.

**Comment by daxian-dbw** [comment]
@iSazonov I'm planning to use preview2 CLI and explicitly specify the `RuntimeFrameworkVersion` to be `preview3` packages. Maybe this is good in the long run so that we don't need to change the CLI we are using when we want to move to later version of .NET Core.

**Comment by iSazonov** [comment]
I believe it is good for us. Will this affect the use of PowerShell Core SDK by third-party developers (for binary modules)?

**Comment by daxian-dbw** [comment]
This issue is fixed by #4144
